components/TextFields/src/TypographyThemer/MDCTextFieldTypographyThemer.m

```diff
@@ -0,0 +1,80 @@
+/*
+ Copyright 2018-present the Material Components for iOS authors. All Rights Reserved.
+
+ Licensed under the Apache License, Version 2.0 (the "License");
+ you may not use this file except in compliance with the License.
+ You may obtain a copy of the License at
+
+ http://www.apache.org/licenses/LICENSE-2.0
+
+ Unless required by applicable law or agreed to in writing, software
+ distributed under the License is distributed on an "AS IS" BASIS,
+ WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ See the License for the specific language governing permissions and
+ limitations under the License.
+ */
+
+#import "MDCTextFieldTypographyThemer.h"
+
+@implementation MDCTextFieldTypographyThemer
+
++ (void)applyTypographyScheme:(id<MDCTypographyScheming>)typographyScheme
+        toTextInputController:(id<MDCTextInputController>)textInputController {
+  textInputController.inlinePlaceholderFont = typographyScheme.subtitle1;
+  textInputController.leadingUnderlineLabelFont = typographyScheme.caption;
+  textInputController.trailingUnderlineLabelFont = typographyScheme.caption;
+  if ([textInputController
+       conformsToProtocol:@protocol(MDCTextInputControllerFloatingPlaceholder)]) {
+    id<MDCTextInputControllerFloatingPlaceholder> textInputControllerFloatingPlaceholder =
+        (id<MDCTextInputControllerFloatingPlaceholder>)textInputController;
+    if (!typographyScheme.subtitle1 ||
+        !typographyScheme.caption ||
```

Comment:
Both of these parameters are non-null, so these checks are redundant.

Reply:
Done.